src/main/java/reposense/report/CommitReportJson.java

```diff
@@ -18,6 +20,27 @@
     private final Map<Author, Float> authorContributionVariance;
     private final Map<Author, String> authorDisplayNameMap;
 
+    /**
+     * Empty constructor to construct an empty commit report.
+     */
+    public CommitReportJson() {
+        Author emptyAuthor = new Author(Author.UNKNOWN_AUTHOR_GIT_ID);
+        authorWeeklyIntervalContributions = new HashMap<>();
+        authorWeeklyIntervalContributions.put(emptyAuthor, Collections.emptyList());
+
+        authorDailyIntervalContributions = new HashMap<>();
+        authorDailyIntervalContributions.put(emptyAuthor, Collections.emptyList());
+
+        authorFinalContributionMap = new HashMap<>();
+        authorFinalContributionMap.put(emptyAuthor, 0);
+
+        authorContributionVariance = new HashMap<>();
+        authorContributionVariance.put(emptyAuthor, (float) 0.0);
+
+        authorDisplayNameMap = new HashMap<>();
+        authorDisplayNameMap.put(emptyAuthor, "FAILED TO CLONE OR CHECKOUT THIS REPOSITORY");
+    }
+
```

Comment:
so are we gonna to remove this part? to actually write out a blank hashmap with no keys?

i can use that to detect when to show the "error" in the UI

Reply:
can refer to @yong24s description above. So the repos in summary.json will now have additional one more field to hold the message.